gammapy/maps/geom.py

```diff
@@ -1366,3 +1366,55 @@ def is_image(self):
             return True
         is_image = len(self.axes) == 0
         return is_image
+
+    @property
+    def axes_names(self):
+        """Returns list of axes names"""
+        return [_.name for _ in self.axes]
+
+    def get_axis_by_name(self, name):
+        """Return axis with corresponding name or type.
+
+        Parameters
+        ----------
+        name : str
+           the name of the requested axis
+
+        Returns
+        -------
+        axis : `~gammapy.maps.MapAxis`
+            the corresponding axis
+
+        """
+
+        # TODO : we implictly assume all axes have different names. This should be enforced at MapGeom creation.
+        for i, axis in enumerate(self.axes):
+            if axis.name.upper() == name.upper():
+                return axis
+        raise ValueError("Cannot find axis named {}".format(name))
+
+    def get_axes_by_type(self, type):
```

Comment:
Is it really useful to try and support maps with multiple axes of a given type?
I would suggest to change to `get_axis_by_type` to get one, for symmetry with `get_axis_by_name` and also it's what you need in the one caller for this functionality.

Note that filtering the axes list by type is still easily possible with a one-liner, should we ever need this.
```
axes = [ax for ax in m.geom.axes if ax.type == 'energy']
```